### DISCUSSION TITLE
Convert Sliced Part back to STEP file for evaluation

### DISCUSSION THREAD

**Discussion by AWACS-SkyKnight** [OrcaSlicer/OrcaSlicer#5008] [Ideas]
Hello
I am a mechanical engineering student and I use 3D Printing a lot for design.

A problem that me and my colleagues have is that there isn't a good way to evaluate 3D printed parts with FEA due to infill, walls, layer adhesion etc. As far as I am aware, there aren't any programs that can reverse gcode back into geometry and doing so would be a major challenge.

However, Orcaslicer, like other slicers, generates a detailed preview of your print. This preview presumably already has a decent sense of the geometry, as options like "make overhangs printable" exist, and layer width and height are defined as well as the location of any given extrusion. 

Would it be possible to introduce an FEA tool that can perform simulations based on a sliced model? Or at least, export the sliced model as a STEP file to then load in a program such as CATIA or Ansys to perform simulations?

I am not too familiar with coding so I'm not sure how difficult a new FEA system would be, but if done in the slicer this could open up major opportunities, such as being able to define layer adhesion as a factor in the analysis, or having a special meshing method which better represents individual layers.

Alternatively, generating a STEP file would likely be much easier to pull off. I think this would make the slicer much more advantageous in engineering terms.

What is the likelihood this feature could become a reality?

**Comment by foreachthing** [comment]
In Prusa, you can export an .obj from the tool paths (gcode viewer).
Blender can then import the .obj and export a .stl. ANSYS can then import the .stl:
![image](https://github.com/SoftFever/OrcaSlicer/assets/10420187/f19b95b9-3738-4f03-8f45-07e12f7839b5)

![image](https://github.com/SoftFever/OrcaSlicer/assets/10420187/e487f7a4-499c-4d48-8755-e6d42e9c44f0)

![image](https://github.com/SoftFever/OrcaSlicer/assets/10420187/527c4be1-6a07-40eb-95c6-87837912e4ea)
